module/localisation/RobotParticleLocalisation/src/RobotModel.h

```diff
@@ -47,42 +62,183 @@ namespace localisation {
 
         RobotModel() {}
 
-        arma::vec::fixed<RobotModel::size> timeUpdate(const arma::vec::fixed<RobotModel::size>& state, double deltaT);
+        StateVec time(const StateVec& state, double deltaT) {
+            Eigen::Matrix<Scalar, RobotModel<Scalar>::size, 1> new_state = state;
+            return new_state;
+        }
 
-        arma::vec predictedObservation(const arma::vec::fixed<RobotModel::size>& state,
-                                       const arma::vec& actual_position,
-                                       const utility::math::matrix::Transform3D& Hcw,
-                                       const message::vision::Goal::MeasurementType& type,
-                                       const message::support::FieldDescription& fd);
+        Eigen::VectorXd predictedObservation(const StateVec& state,
+                                             const Eigen::VectorXd& actual_position,
+                                             const Eigen::Affine3d& Hcw,
+                                             const message::vision::Goal::MeasurementType& type,
+                                             const message::support::FieldDescription& fd) {
+
+            double c = std::cos(state[2]);
+            double s = std::sin(state[2]);
+            Eigen::Matrix<Scalar, 3, 3> rotation;
+            rotation << c << -s << 0 << s << c << 0 << 0 << 0 << 1;
+            Eigen::Affine3d Hfw;
+            Hfw.translation()   = Eigen::Vector3d{state[0], state[1], 0};
+            Hfw.linear()        = rotation;
+            Eigen::Affine3d Hcf = Hcw * Hfw.inverse();
+
+            // rZFf = vector from field origin to zenith high in the sky
+            Eigen::Vector4d rZFf = {0, 0, 1, 0};
+            rZFf                 = Hcf * rZFf;
+
+            if (type == Goal::MeasurementType::CENTRE) {
+                // rGCc = vector from camera to goal post expected position
+                Eigen::Vector4d rGCc_4   = {actual_position[0], actual_position[1], actual_position[2], 1};
+                rGCc_4                   = Hcf * rGCc_4;
+                Eigen::Vector3d rGCc     = {rGCc_4[0], rGCc_4[1], rGCc_4[2]};
+                Eigen::Vector3d rGCc_sph = cartesianToSpherical(rGCc);  // in r,theta,phi
```

Comment:
```suggestion
                Eigen::Vector4d rGCc_4(actual_position.x(), actual_position.y(), actual_position.z(), 1);
                rGCc_4                   = Hcf * rGCc_4;
                Eigen::Vector3d rGCc(rGCc_4.x(), rGCc_4.y(), rGCc_4.z());
                Eigen::Vector3d rGCc_sph = cartesianToSpherical(rGCc);  // in r,theta,phi
```

Similar for the rest